kavaplugin/src/main/java/com/kaltura/playkit/plugins/kava/AverageBitrateCounter.java

```diff
@@ -0,0 +1,89 @@
+package com.kaltura.playkit.plugins.kava;
+
+
+import com.kaltura.playkit.PKLog;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+/**
+ * Created by anton.afanasiev on 08/02/2018.
+ */
+
+public class AverageBitrateCounter {
+
+    private PKLog log = PKLog.get(AverageBitrateCounter.class.getSimpleName());
+
+    private boolean shouldCount = false;
+    private long currentTrackBitrate = -1;
+    private long currentTrackStartTimestamp = 0;
+
+    private Map<Long, Long> averageTrackPlaybackDuration = new HashMap<>();
+
+    /**
+     * Calculate average bitrate for the entire media session.
+     * @param totalPlaytimeSum - total amount of the player being in active playback mode.
+     * @return - average bitrate.
+     */
+    long getAverageBitrate(long totalPlaytimeSum) {
+        if (!shouldCount) {
+            log.w("Should not try to get average bitrate while player is paused");
+        }
+
+        updateBitratePlayTime();
+        Iterator iterator = averageTrackPlaybackDuration.entrySet().iterator();
+        long averageBitrate = 0;
+        while (iterator.hasNext()) {
+            Map.Entry pair = (Map.Entry)iterator.next();
+            long bitrate = (long) pair.getKey();
+            long playTime = (long) pair.getValue();
+            averageBitrate += (bitrate * playTime) / totalPlaytimeSum;
+
```

Comment:
make sure totalPlaytimeSum != 0

Reply:
totalPlaytimeSum can`t be 0. getAverage is requested from VIEW event which means that there was at least 10 seconds of active playback.